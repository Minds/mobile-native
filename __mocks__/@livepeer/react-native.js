// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
