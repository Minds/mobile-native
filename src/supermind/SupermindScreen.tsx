import { RouteProp, useNavigation } from '@react-navigation/core';
import { observer } from 'mobx-react';
import React from 'react';
import { ScrollView } from 'react-native';
import { showNotification } from '../../AppMessages';
import CenteredLoading from '../common/components/CenteredLoading';
import useApiFetch from '../common/hooks/useApiFetch';
import { Screen, ScreenHeader } from '../common/ui';
import { AppStackParamList } from '../navigation/NavigationTypes';
import SupermindRequest from './SupermindRequest';
import SupermindRequestModel from './SupermindRequestModel';
import { withErrorBoundaryScreen } from '~/common/components/ErrorBoundaryScreen';

interface SupermindScreenProps {
  route: RouteProp<AppStackParamList, 'Supermind'>;
}

export default withErrorBoundaryScreen(
  observer(function SupermindScreen({ route }: SupermindScreenProps) {
    const navigation = useNavigation();
    const { supermindRequest, guid } = route.params ?? {};
    let { supermind, loading, error } = useSupermind(guid);

    if (!supermind && supermindRequest) {
      supermind = SupermindRequestModel.checkOrCreate(supermindRequest);
    }

    if (error && !supermind) {
      showNotification('Supermind not found', 'danger');
      navigation.goBack();
    }

    return (
      <Screen safe>
        <ScrollView>
          <ScreenHeader title="Supermind" back />
          {supermind ? (
            <SupermindRequest request={supermind} />
          ) : (
            loading && <CenteredLoading />
          )}
        </ScrollView>
      </Screen>
    );
  }),
  'SupermindScreen',
);

const useSupermind = (guid: string) => {
  const store = useApiFetch<SupermindRequestModel>('/api/v3/supermind/' + guid);

  return {
    ...store,
    supermind: store.result?.[0]
      ? SupermindRequestModel.create(store.result[0])
      : undefined,
  };
};
