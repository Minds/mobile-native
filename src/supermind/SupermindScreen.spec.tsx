import { render, screen } from '@testing-library/react-native';
import * as React from 'react';
import supermindRequestFaker from '../../__mocks__/fake/supermind/SupermindRequestFaker';
import apiService from '../common/services/api.service';
import SupermindScreen from './SupermindScreen';

jest.mock('../common/services/api.service');
jest.mock('./SupermindRequest', () => () => 'SupermindRequest');

const guid = 'fakeGuid';

describe('SupermindScreen', () => {
  test('should render correctly', () => {
    render(
      <SupermindScreen
        route={
          {
            params: {
              guid,
            },
          } as any
        }
      />,
    );
    apiService.get.mockResolvedValue(supermindRequestFaker());
    expect(apiService.get).toHaveBeenCalledWith(
      '/api/v3/supermind/' + guid,
      expect.anything(),
      expect.anything(),
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
