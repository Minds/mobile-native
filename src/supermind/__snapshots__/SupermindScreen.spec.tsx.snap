// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupermindScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#1F252C",
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginRight": 8,
                  "position": "relative",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 9.333333333333332,
                    "left": 9.333333333333332,
                    "right": 9.333333333333332,
                    "top": 9.333333333333332,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 28,
                          "justifyContent": "center",
                          "overflow": "hidden",
                          "width": 28,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#AEB0B8",
                            "fontSize": 28,
                          },
                          Object {},
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Û∞ÖÅ
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Bold",
                  },
                  Object {
                    "fontSize": 24,
                    "lineHeight": 32,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Supermind
            </Text>
          </View>
          <View />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "minHeight": 100,
          }
        }
      >
        <ActivityIndicator
          color="#1B85D6"
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
